src/index.js

```diff
@@ -1165,157 +1293,181 @@ const _startRenderLoop = () => {
-        return;
-      }
-
-      vrPresentState.hasPose = true;
       if (args.performance) {
         const now = Date.now();
         const diff = now - timestamps.last;
         timestamps.wait += diff;
         timestamps.total += diff;
         timestamps.last = now;
       }
-      _normalizeMatrixArray(localFloat32Array);
-      _normalizeMatrixArray(localFloat32Array2);
-      _normalizeMatrixArray(localFloat32Array3);
-
-      // build xr state
-      const hmdMatrix = localMatrix.fromArray(localFloat32Array);
-
-      hmdMatrix.decompose(localVector, localQuaternion, localVector2);
-      localVector.toArray(xrState.position);
-      localQuaternion.toArray(xrState.orientation);
-
-      hmdMatrix.getInverse(hmdMatrix);
-
-      vrPresentState.system.GetEyeToHeadTransform(0, localFloat32Array4);
-      localMatrix2.fromArray(localFloat32Array4);
-      localMatrix2.decompose(localVector, localQuaternion, localVector2);
-      localVector.toArray(xrState.leftOffset);
-      localMatrix2
-        .getInverse(localMatrix2)
-        .multiply(hmdMatrix);
-      localMatrix2.toArray(xrState.leftViewMatrix);
-
-      vrPresentState.system.GetProjectionMatrix(0, xrState.depthNear[0], xrState.depthFar[0], localFloat32Array4);
-      _normalizeMatrixArray(localFloat32Array4);
-      xrState.leftProjectionMatrix.set(localFloat32Array4);
-
-      vrPresentState.system.GetProjectionRaw(0, localFovArray);
-      for (let i = 0; i < localFovArray.length; i++) {
-        xrState.leftFov[i] = Math.atan(localFovArray[i]) / Math.PI * 180;
-      }
 
-      vrPresentState.system.GetEyeToHeadTransform(1, localFloat32Array4);
-      _normalizeMatrixArray(localFloat32Array4);
-      localMatrix2.fromArray(localFloat32Array4);
-      localMatrix2.decompose(localVector, localQuaternion, localVector2);
-      localVector.toArray(xrState.rightOffset);
-      localMatrix2
-        .getInverse(localMatrix2)
-        .multiply(hmdMatrix);
-      localMatrix2.toArray(xrState.rightViewMatrix);
-
-      vrPresentState.system.GetProjectionMatrix(1, xrState.depthNear[0], xrState.depthFar[0], localFloat32Array4);
-      _normalizeMatrixArray(localFloat32Array4);
-      xrState.rightProjectionMatrix.set(localFloat32Array4);
-
-      vrPresentState.system.GetProjectionRaw(1, localFovArray);
-      for (let i = 0; i < localFovArray.length; i++) {
-        xrState.rightFov[i] = Math.atan(localFovArray[i]) / Math.PI * 180;
-      }
+      if (vrPresentState.oculusSystem) {
+        vrPresentState.oculusSystem.GetPose(
+          localPositionArray3,   // hmd position
+          localQuaternionArray4, // hmd orientation
+          localFloat32Array,     // left eye view matrix
+          localFloat32Array2,    // left eye projection matrix
+          localFloat32Array3,    // right eye view matrix
+          localFloat32Array4     // right eye projection matrix
+        );
+        vrPresentState.hasPose = true;
+
+        xrState.positon = localPositionArray3;
```

Comment:
Typo?

Reply:
I don't think this is a typo. Can you elaborate?

Reply:
posit`I`on?

Reply:
good catch